src/platform/datapath_raw.h

```diff
@@ -256,8 +256,13 @@ typedef struct CXPLAT_SOCKET {
     void* CallbackContext;
     QUIC_ADDR LocalAddress;
     QUIC_ADDR RemoteAddress;
-    BOOLEAN Wildcard;   // Using a wildcard local address. Optimization to avoid always reading LocalAddress.
-    BOOLEAN Connected;  // Bound to a remote address
+    BOOLEAN Wildcard;           // Using a wildcard local address. Optimization
+                                // to avoid always reading LocalAddress.
+    BOOLEAN Connected;          // Bound to a remote address
+    uint8_t CibirIdLength;      // Value of 0 indicates CIBIR isn't used
+    uint8_t CibirIdOffsetSrc;
+    uint8_t CibirIdOffsetDst;
+    uint8_t CibirId[6];
```

Comment:
>CibirId[6]; [](http://example.com/codeflow?start=12&length=11)

Would it be better to define CibirId in a struct like this?
struct {
    uint8_t Length;
    uint8_t Offset;
    uint8_t Data[4];
};

Reply:
Here, the CibirId is purely the ID, no encoded values internally.

Reply:
Hmm, that confused me because we name it CibirId in QUIC_CONNECTION.

Reply:
In the QUIC layer, we encode everything in a contiguous memory block for efficiency/packing. Here the space isn't as much an issue so it's unpacked.

Reply:
It's mainly the naming that confuses me. It's inconsistent across the entire code base. Maybe rename it to CibirIdData? or at least add a comment - "// CIBIR ID data" or maybe just using the macro QUIC_MAX_CIBIR_LENGTH would make it more clearer.